src/format/scan.rs

```diff
@@ -463,6 +467,120 @@ mod tests {
         assert_eq!(space("a "), Err(INVALID));
     }
 
+    #[test]
+    fn test_timezone_name_skip() {
+        let s = "\r";
+        let ans = timezone_name_skip(s);
+        assert!(ans.is_ok());
+    }
+
+    #[test]
+    fn test_timezone_offset_2822() {
+        let s = "cSt";
+        let ans = timezone_offset_2822(s);
+        assert_eq!(format!("{:?}", ans), "Ok((\"\", Some(-21600)))");
+
+        let s = "pSt";
+        let ans = timezone_offset_2822(s);
+        assert_eq!(format!("{:?}", ans), "Ok((\"\", Some(-28800)))");
+
+        let s = "mSt";
+        let ans = timezone_offset_2822(s);
+        assert_eq!(format!("{:?}", ans), "Ok((\"\", Some(-25200)))");
+
+        let s = "Gp";
+        let ans = timezone_offset_2822(s);
+        assert_eq!(format!("{:?}", ans), "Ok((\"\", None))");
+
+        let s = "-1551";
+        let ans = timezone_offset_2822(s);
+        assert_eq!(format!("{:?}", ans), "Ok((\"\", Some(-57060)))");
+    }
+
+    #[test]
+    fn test_short_or_long_month0() {
+        let s = "JUn";
```

Comment:
I'd prefer to shorten these test cases by avoiding binding `s`. Instead, just have `let ans = short_or_long_month0("JUn");` instead. (Though honestly it would also be helpful if the `ans` binding name was a bit more semantically meaningful in context.)